package/MDAnalysis/topology/TOPParser.py

```diff
@@ -563,9 +598,23 @@ def parsesection_mapper(self, numlines, mapper):
             A list of all entries in a given parm7 section
         """
         section = []
-        y = next(self.topfile).strip("%FORMAT(")
-        y.strip(")")
-        x = FORTRANReader(y)
+
+        def get_fmt(file):
+            line = next(file)
+            if line[:7] == "%FORMAT":
+                return line[8:].split(")")[0]
+            elif line[:8] == "%COMMENT":
+                # Skip %COMMENT lines until finding the %FORMAT statement.
+                return get_fmt(file)
+            else:
+                raise ValueError(
```

Comment:
It would be great to make sure these are tested, you can either patch in the methods or create small fake parm7 files.

Reply:
Done.